src/librustc/mir/mod.rs

```diff
@@ -212,26 +210,53 @@ impl<'tcx> Body<'tcx> {
 
     #[inline]
     pub fn basic_blocks_mut(&mut self) -> &mut IndexVec<BasicBlock, BasicBlockData<'tcx>> {
-        self.cache.invalidate();
+        self.predecessors_cache = None;
+//        self.cache.invalidate();
```

Comment:
@oli-obk do you think I should add a fixme or todo comment here about improving the cache invalidation?

Reply:
either that or open an issue